velox/functions/sparksql/RegisterArithmetic.cpp

```diff
@@ -81,6 +81,13 @@ void registerArithmeticFunctions(const std::string& prefix) {
   registerFunction<HypotFunction, double, double, double>({prefix + "hypot"});
   registerFunction<sparksql::Log2Function, double, double>({prefix + "log2"});
   registerFunction<sparksql::Log10Function, double, double>({prefix + "log10"});
+  registerFunction<
+      WidthBucketFunction,
+      int64_t,
+      double,
+      double,
+      double,
+      int64_t>({prefix + "width_bucket"});
```

Comment:
Would you document this function in https://github.com/facebookincubator/velox/tree/main/velox/docs/functions/spark?

It looks like Spark has a nice one for reference https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala#L1792-L1803.

Reply:
refactor to align how spark implement this function